### ISSUE TITLE
kfctl should fail if an overlay is not found

### ISSUE THREAD

**Issue by yanniszark** [kubeflow/kfctl#155]
If an overlay is not found, this indicates a user error and kfctl should not continue, but fail loudly.
https://github.com/kubeflow/kfctl/blob/9957ea14f901f3de107c059561e1de0a8529b497/pkg/kfapp/kustomize/kustomize.go#L867

We should change this to `log.Fatal` or return an error.

**Comment by jlewi** [comment]
I don't think we should just crash because kfctl could be used programmatically; e.g in the server.
But throwing an error seems like a sensible thing todo.
/cc @gabrielwen
